### PR TITLE
CI: enable github workflows

### PR DESCRIPTION
**Pull request by emturner**
Enable CI for builds/PRs.

Switches away from `actions-rs` as this is unmaintained. See [thread](https://github.com/actions-rs/toolchain/issues/216)

### CODE REVIEW

**Comment by sribst** [comment]
I'm definitly not a CI expert, but LGTM from what I understand !

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -18,19 +18,15 @@ jobs:
...
         if: matrix.sanitizer != 'address'
-        uses: actions-rs/toolchain@v1
-        with:
-          toolchain: 1.58.1
-          default: true
+        uses: dtolnay/rust-toolchain@1.58.1
       - name: Setup Rust (for sanitizer)
         if: matrix.sanitizer == 'address'
-        uses: actions-rs/toolchain@v1
+        uses: dtolnay/rust-toolchain@master
```
Comment (acerone85) [inline .github/workflows/build.yml:27]: why @master here?

**Review by acerone85 — APPROVED** [review]
I have checked that the checks performed by the CI are indeed using the updated files. 
As far as i understand action-rs/toolchain became unmantained, and dtolnay/rust-toolchain is being used instead.

LGTM

File: .github/workflows/build.yml (hunk 1 above)
Comment (emturner) [reply .github/workflows/build.yml:27]: This is due to the use of the `nightly` toolchain - there's no specific `rev` for each version of nightly

**Pull request merged**